cps/db.py

```diff
@@ -338,37 +338,9 @@ def __init__(self):
         threading.Thread.__init__(self)
         self.engine = None
         self.session = None
-        self.queue = None
         self.log = None
         self.config = None
 
-    def add_queue(self,queue):
-        self.queue = queue
-        self.log = logger.create()
-
-    def run(self):
-        while True:
-            i = self.queue.get()
-            if i == 'dummy':
-                self.queue.task_done()
-                break
-            if i['task'] == 'add_format':
-                cur_book = self.session.query(Books).filter(Books.id == i['id']).first()
-                cur_book.data.append(i['format'])
-                try:
-                    # db.session.merge(cur_book)
-                    self.session.commit()
-                except OperationalError as e:
-                    self.session.rollback()
-                    self.log.error("Database error: %s", e)
-                    # self._handleError(_(u"Database error: %(error)s.", error=e))
-                    # return
-            self.queue.task_done()
-
-
-    def stop(self):
-        self.queue.put('dummy')
-
```

Comment:
All queue stuff was removed from CalibreDB class due to #1565 being merged in, which allows the database to be accessed safely from other threads. I've tested with adding converted formats via the Convert task, and can confirm that, at least with my many tests, adding it directly in the Worker thread works 100%